rest/adminapitest/admin_api_test.go

```diff
@@ -4015,3 +4015,112 @@ func TestDatabaseCreationErrorCode(t *testing.T) {
 		rest.RequireStatus(t, resp, http.StatusPreconditionFailed)
 	}
 }
+
+// TestDatabaseCreationWithEnvVariable:
+//   - Create rest tester that enables admin auth and disallows db config env vars
+//   - Create CBS user to authenticate with over admin port to force auth scope callback call
+//   - Create db with sync function that calls env variable
+//   - Assert that db is created
+func TestDatabaseCreationWithEnvVariable(t *testing.T) {
```

Comment:
This test doesn't work for `SG_TEST_USE_GSI=false` or `SG_TEST_USE_DEFAULT_COLLECTION=true`

Will need to build the db config to accommodate those modes.

Some existing functions to consider using or adapting:
- `dbConfigForTestBucket`
- `makeDbConfig`

Should be able to include string values like `"${VAR}"` for sync function before marshalling it to pass into the request.

Reply:
Oh wow didn't even think about that, on it fixing it